--- c2klisp-min.c	2026-05-07 20:18:42.811225153 -0500
+++ c2klisp-min.c	2026-05-07 20:18:29.395685209 -0500
@@ -45,17 +45,8 @@
 	X("\002if", sym_if) \
 	X("\006lambda", sym_lambda) \
 	X("\006define", sym_define) \
-	X("\004eval", sym_eval) \
 	X("\006expand", sym_expand) \
-	X("\006gensym", sym_gensym) \
-	X("\005print", sym_print) \
-	X("\001+", sym_add) \
-	X("\001-", sym_sub) \
-	X("\001*", sym_mul) \
-	X("\001/", sym_div) \
-	X("\001%", sym_mod) \
-	X("\001>", sym_grt) \
-	X("\001=", sym_equ)
+	X("\006gensym", sym_gensym)
 
 // Declare character pointer variables for each built-in symbol
 #define DECLARE_SYMVAR(sym, id) char *id;
@@ -65,7 +56,6 @@
 #define ERROR     ((void *)1)  // used as a generic error value
 #define FORWARD   ((void *)2)  // used for signaling that a cell has already been copied by garbage collection
 #define CONTINUE  ((void *)3)  // used for signaling that an expression should be tail call optimized
-#define NUMBER    ((void *)4)  // used for signaling that a cell's cdr has a raw numeric value
 
 
 // **************** Memory regions and region-based type inference ****************
@@ -102,7 +92,7 @@
 
 static inline bool atom(void *l)
 {
-	return !IN(l, cells) || *CAR(l) == NUMBER;
+	return !IN(l, cells);
 }
 
 static inline void *car(void *l)
@@ -131,23 +121,12 @@
 #define cadr(x) car(cdr(x))
 #define cdar(x) cdr(car(x))
 
-// Number-related definitions
-#define NUM intptr_t
-#define NUM_FMT "%ld"
-#define ISNUM(x) (IN((x),cells) && *CAR(x) == NUMBER)
-#define UNWRAP(x) (NUM)(*CDR(x))
-#define WRAP(x) cons(NUMBER, (void *)(x))
-
 // Value printing
 void print(void *x)
 {
 	if (!x) {
 		printf("()");
 	} else if (IN(x, cells)) {
-		if (*CAR(x) == NUMBER) {
-			printf(NUM_FMT, UNWRAP(x));
-			return;
-		}
 		// For lists, first print the head
 		printf("(");
 		print(car(x));
@@ -268,11 +247,6 @@
 		return ERROR;
 	*s = next_sym - s;
 	*(++next_sym) = '\0';
-	NUM n;
-	if (sscanf(&s[1], NUM_FMT, &n) > 0) {
-		next_sym = s;
-		return WRAP(n);
-	}
 	return intern(s);
 }
 
@@ -317,8 +291,8 @@
 	if (*CAR(x) == FORWARD) // No need to copy values that have already been copied
 		return cdr(x);
 	// Deep copy the value normally
-	void *a = *CAR(x) == NUMBER ?  NUMBER : copy(car(x), diff);
-	void *d = *CAR(x) == NUMBER ? *CDR(x) : copy(cdr(x), diff);
+	void *a = copy(car(x), diff);
+	void *d = copy(cdr(x), diff);
 	void *res = (void **)cons(a, d) - diff;
 	// Leave a forward pointer to indicate that the cell has already been copied
 	*CAR(x) = FORWARD;
@@ -417,34 +391,6 @@
 	if (car(x) == sym_quote) // quote
 		return cadr(x);
 
-	// Numeric primitives
-#define TOBOOL(x) ((x) ? sym_t : NULL)
-#define BINOP(sym, conv, op, err) \
-	if (car(x) == sym) { \
-		void *a = eval(cadr(x), env); \
-		void *b = eval(caddr(x), env); \
-		if (ISNUM(a) && ISNUM(b)) \
-			return conv(UNWRAP(a) op UNWRAP(b)); \
-		return err; \
-	}
-	BINOP(sym_add,   WRAP,  +, NULL)
-	BINOP(sym_sub,   WRAP,  -, NULL)
-	BINOP(sym_mul,   WRAP,  *, NULL)
-	BINOP(sym_div,   WRAP,  /, NULL)
-	BINOP(sym_mod,   WRAP,  %, NULL)
-	BINOP(sym_grt, TOBOOL,  >, NULL)
-	BINOP(sym_equ, TOBOOL, ==, NULL)
-	BINOP(sym_eq,  TOBOOL, ==, TOBOOL(a == b))
-	// ^ Override `eq` to handle numbers
-
-	// Print function - useful for debugging
-	if (car(x) == sym_print) {
-		void *a = eval(cadr(x), env);
-		print(a);
-		printf("\n");
-		return a;
-	}
-
 	// Handle primitive functions
 	if (car(x) == sym_car) // car
 		return car(eval(cadr(x), env));
@@ -476,9 +422,7 @@
 	// Trampoline
 	void *ret;
 	while ((ret = eval_base(x, env)) == CONTINUE) {
-		if (car(x) == sym_eval) // eval -> continue from expression given by evaluated argument
-			x = eval(cadr(x), env);
-		else if (car(x) == sym_if) // if -> continue from expression switched by condition
+		if (car(x) == sym_if) // if -> continue from expression switched by condition
 			x = eval(cadr(x), env) ? caddr(x) : cadddr(x);
 		else // must be a function application -> continue from apply (lambda body / fexpr result)
 			x = apply(eval(car(x), env), cdr(x), &env);
